database/engine/metadata_log/metalogpluginsd.c

```diff
@@ -116,41 +125,55 @@ PARSER_RC metalog_pluginsd_context_action(void *user, uuid_t *uuid)
 
     ret = find_object_by_guid(uuid, object, 49);
     switch (ret) {
-    case GUID_TYPE_CHAR:
-        fatal_assert(0);
-        break;
-    case GUID_TYPE_CHART:
-    case GUID_TYPE_DIMENSION:
-        host = ctx->rrdeng_ctx->host;
-        switch (ret) {
-        case GUID_TYPE_CHART:
-            chart_char_guid = (uuid_t *)(object + 16);
-
-            ret = find_object_by_guid(chart_char_guid, id_str, RRD_ID_LENGTH_MAX + 1);
-            fatal_assert(GUID_TYPE_CHAR == ret);
-            ((PARSER_USER_OBJECT *) user)->st = rrdset_find(host, id_str);
+        case GUID_TYPE_NOTFOUND:
+            if (unlikely(ctx->rrdeng_ctx->host && uuid_compare(ctx->rrdeng_ctx->host->host_uuid, *uuid)))
+                error_with_guid(uuid, "Failed to find valid context");
+            break;
+        case GUID_TYPE_CHAR:
+            error_with_guid(uuid, "Ignoring unexpected type GUID_TYPE_CHAR");
             break;
+        case GUID_TYPE_CHART:
         case GUID_TYPE_DIMENSION:
-            chart_guid = (uuid_t *)(object + 16);
-
-            ret = find_object_by_guid(chart_guid, chart_object, 33);
-            fatal_assert(GUID_TYPE_CHART == ret);
-            chart_char_guid = (uuid_t *)(chart_object + 16);
-
-            ret = find_object_by_guid(chart_char_guid, id_str, RRD_ID_LENGTH_MAX + 1);
-            fatal_assert(GUID_TYPE_CHAR == ret);
-            ((PARSER_USER_OBJECT *) user)->st = rrdset_find(host, id_str);
+            host = ctx->rrdeng_ctx->host;
+            switch (ret) {
```

Comment:
Are you putting a `switch(ret)` inside a shared case in a `switch(ret)` to avoid repeating the `host = ` line above ??

Reply:
This is a remnant from the persistent metadata feature branch. It used to have a lot of code about hosts but it got cleaned up and the monstrosity remained.